transform-plugins/src/main/java/io/cdap/plugin/XMLParser.java

```diff
@@ -291,26 +309,32 @@ public Config(String inputField, String encoding, String xPathFieldMapping, Stri
      *
      * @return output schema
      */
-    private Schema getOutputSchema() {
+    private Schema getOutputSchema(FailureCollector collector) {
       List<Schema.Field> fields = new ArrayList<>();
+      List<String> fieldNames = new ArrayList<>();
       String[] mappings = fieldTypeMapping.split(",");
       for (String mapping : mappings) {
         String[] params = mapping.split(":");
         String fieldName = params[0].trim();
         if (Strings.isNullOrEmpty(fieldName)) {
-          throw new IllegalArgumentException("Field name cannot be null or empty.");
+          collector.addFailure("Field name cannot be null or empty.", null)
+              .withConfigElement(FIELD_TYPE_MAPPING, mapping);
         } else if (params.length < 2 || Strings.isNullOrEmpty(params[1])) {
-          throw new IllegalArgumentException("Type cannot be null. Please specify type for " + fieldName);
-        }
-        Schema.Field field = Schema.Field.of(fieldName, Schema.nullableOf(Schema.of(Schema.Type.valueOf(
-          params[1].trim().toUpperCase()))));
-        if (fields.contains(field)) {
-          throw new IllegalArgumentException(String.format("Field %s already has type specified. Duplicate field %s",
-                                                           fieldName, fieldName));
+          collector.addFailure(String.format("A type must be provided for field '%s'.", fieldName), null)
+              .withConfigElement(FIELD_TYPE_MAPPING, mapping);
         } else {
-          fields.add(field);
+          Schema.Field field = Schema.Field.of(fieldName, Schema.nullableOf(Schema.of(Schema.Type.valueOf(
```

Comment:
When there's so many function calls within function calls, it gets hard to read. Break this up into 2 statements.

Reply:
Fixed